core/usage_interceptor.go

```diff
@@ -247,14 +255,17 @@ func (t *Textile) postUsageFunc(ctx context.Context, method string) error {
 		"/api.bucketsd.pb.APIService/Remove",
 		"/api.bucketsd.pb.APIService/RemovePath",
 		"/api.bucketsd.pb.APIService/PushPathAccessRoles":
+		// If the main context is canceled, we still want to be able to record usage.
+		sctx, cancel := context.WithTimeout(util.NewClonedContext(ctx), time.Second*10)
+		defer cancel()
 		if _, err := t.bc.IncCustomerUsage(
-			ctx,
+			sctx,
 			account.Owner().Key,
 			map[string]int64{
 				"stored_data": owner.StorageDelta,
 			},
 		); err != nil {
-			return err
+			log.Infof("postUsageFunc: %v", err)
```

Comment:
There's no point in returning a usage error to the user here since they have already "used" the resources. They will get kicked out during the next request.